packages/console/src/components/RoleUsersTransfer/components/SourceUsersBox/index.tsx

```diff
@@ -0,0 +1,99 @@
+import type { User } from '@logto/schemas';
+import { conditional } from '@silverhand/essentials';
+import type { ChangeEvent } from 'react';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import useSWR from 'swr';
+
+import Search from '@/assets/images/search.svg';
+import Pagination from '@/components/Pagination';
+import TextInput from '@/components/TextInput';
+import { defaultPageSize } from '@/consts';
+import type { RequestError } from '@/hooks/use-api';
+import useDebounce from '@/hooks/use-debounce';
+import { formatKeyword } from '@/hooks/use-table-search-params';
+import * as transferLayout from '@/scss/transfer.module.scss';
+import { buildUrl } from '@/utilities/url';
+
+import SourceUserItem from '../SourceUserItem';
+import * as styles from './index.module.scss';
+
+type Props = {
+  roleId: string;
+  onChange: (value: User[]) => void;
+  selectedUsers: User[];
+};
+
+const searchDelay = 500;
+
+const SourceUsersBox = ({ roleId, selectedUsers, onChange }: Props) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const [pageIndex, setPageIndex] = useState(1);
+  const [keyword, setKeyword] = useState('');
+  const debounce = useDebounce();
```

Comment:
nit: I think `useDebounce` can accept a timeout param, and given a default value 500ms. Then you don't have to define the global variable `searchDelay` any more

Reply:
Good call. I will update the hook in the next PR.